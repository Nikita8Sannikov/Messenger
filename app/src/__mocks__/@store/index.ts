import useAuthStore from "./auth";
import useGlobalStore from "./global";
import useImagesCarouselStore from "./images-carousel";
import usePhotosStore from "./photos";
import useProfileStore from "./profile";
import useUIStore from "./ui";
import useUserStore from "./user";

const resetAllStores = jest.fn(() => {
	useUserStore.getState().reset();
	useGlobalStore.getState().reset();
	useUIStore.getState().reset();
	useImagesCarouselStore.getState().reset();
	useAuthStore.getState().reset();
	useProfileStore.getState().reset();
	usePhotosStore.getState().reset();
});

export default resetAllStores; 