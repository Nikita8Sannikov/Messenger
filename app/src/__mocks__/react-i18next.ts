export const useTranslation = () => ({
    t: (key: string) => {
        const dictionary: Record<string, string> = {
            "menu.profile": " My profile",
            "menu.messanger": "Messenger",
            "menu.friends": "Friends",
            "menu.to-developers": "To developers",
            "header.settings": "Settings",
            "header.help": "Help",
            "header.logout": "Log Out"
        };
        return dictionary[key];
    },
    i18n: {
        changeLanguage: jest.fn(),
        use: jest.fn().mockReturnThis(), // Мок для вызова use()
        init: jest.fn(), // Мок для вызова init()
    },
})

export const initReactI18next = {
    type: "3rdParty",
    init: jest.fn(),
}