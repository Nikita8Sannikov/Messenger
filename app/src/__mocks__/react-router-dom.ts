import { createElement, type ReactNode } from "react";
import * as ReactRouterDom from "react-router-dom";

const mockNavigate = jest.fn();
const mockLocation = {
	pathname: "/",
	search: "",
	hash: "",
	state: null,
	key: "default",
};

const mockParams = {};

const useNavigate = () => mockNavigate;
const useLocation = () => mockLocation;
const useParams = () => mockParams;

const matchPath = (pattern: string, pathname: string) => {
	if (pattern === pathname) {
		return {
			params: {},
			pathname,
			pattern,
			pathnameBase: pathname,
		};
	}
	return null;
};

const Routes = ({ children }: { children: ReactNode }) => children;
const Route = ({ children }: { children: ReactNode }) => children;
const Navigate = ({ to }: { to: string }) => createElement("div", { "data-testid": "navigate", "data-to": to });

export { useNavigate, mockNavigate, useLocation, mockLocation, useParams, mockParams, matchPath, Routes, Route, Navigate };
export default {
	...ReactRouterDom,
	useNavigate,
	useLocation,
	useParams,
	matchPath,
	Routes,
	Route,
	Navigate,
}; 