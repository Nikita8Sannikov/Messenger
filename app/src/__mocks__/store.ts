// import { configureStore } from "@reduxjs/toolkit";

// export const mockStore = configureStore({ reducer: () => ({}) });
