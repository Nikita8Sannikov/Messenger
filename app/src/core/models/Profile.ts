import EventEmitter from "eventemitter3";
import { Dispatch, SetStateAction } from "react";
import dayjs from "dayjs";

import { User } from "@core/models/User";
import { IFormValues } from "@pages/Edit";
import { IPhoto } from "@custom-types/models.types";

// Контракт модели "Профиль пользователя"
export interface Profile extends EventEmitter {
    user: User;

    onClickAvatar: () => void;
    onDeleteAvatar: (setLoading?: Dispatch<SetStateAction<boolean>>) => void;
    onSetAvatar: (newAvatar: { id: string; newAvatarUrl: string; newPhotoUrl: string; }) => void;
    getAllPhotos: (setLoading: Dispatch<SetStateAction<boolean>>) => void;
    updatePhotosCount: (count: number) => void;
    onClickPhoto: (photos: IPhoto[], index: number) => void;
    addPhotos: (data: Object, setLoading: Dispatch<SetStateAction<boolean>>) => void;
    deletePhoto: (data: Object, photos: IPhoto[], path: string) => void;
    getFriends: (setLoading: Dispatch<SetStateAction<boolean>>) => void;
    editInfo: ( result: IFormValues) => void;
}