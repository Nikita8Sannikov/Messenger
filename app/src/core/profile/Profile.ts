import EventEmitter from "eventemitter3";

import Request from "../Request";
import { setImagesInCarousel, setLoading } from "../../state/main/slice";
import { addPhotos, changeUserField, deletePhoto, setPhotos, setPhotosCount, setUser, setUserDetail } from "../../state/user/slice";
import { setFriendsCount, setSubscribersCount, setTopFriends } from "../../state/friends/slice";
import { ApiRoutes } from "../../types/enums";
import { IPhoto, IUser, IUserDetails } from "../../types/models.types";
import { AppDispatch } from "../../types/redux.types";
import { MainClientEvents } from "../../types/events";
import { MY_ID, NO_PHOTO } from "../../utils/constants";
import { currentDate } from "../../utils/datetime";
import { AVATAR_URL } from "../../utils/files";
import { getFullName } from "../../utils";

// Класс, отвечающий за полный объект пользователя
export default class Profile extends EventEmitter {
    private _user!: IUser;

    constructor(private readonly _userId: string, private readonly _request: Request, private readonly _dispatch: AppDispatch) {
        super();

        if (this._userId === MY_ID) {
            this._getMe();
        } else {
            this._getUser();
        }
    }

    get id(): string {
        return this._user.id;
    }

    get user(): IUser {
        return this._user;
    }

    get fullName(): string {
        return this._user.firstName + " " + this._user.thirdName;
    }

    get avatarUrl(): string {
        return this._user.avatarUrl ? this._user.avatarUrl : NO_PHOTO;
    }

    // Получение объекта пользователя (себя)
    private _getMe() {
        this._request.get({
            route: ApiRoutes.getUser,
            setLoading: (loading: boolean) => this._dispatch(setLoading(loading)),
            successCb: (data: { success: boolean, user: IUser }) => {
                this._user = data.user;
                this._dispatch(setUser(this._user));
                console.log("Подгрузили инфу о себе: ", this._user);
                this.emit(MainClientEvents.GET_ME);
            }
        });
    }

    // Получение объекта другого пользователя при переходе на его страницу
    private _getUser() {

    }

    // Обновление объекта пользователя в классе Profile и в сторе
    private _changeField(field: string, value: string) {
        this._user = {
            ...this._user,
            [field]: value
        }
        this._dispatch(changeUserField({ field, value }));
    }

    //-------------------------------------------------
    // Методы главной фотографии (аватара)
    //-------------------------------------------------

    // Клик по аватару
    public onClickAvatar(): void {
        this._dispatch(setImagesInCarousel({
            images: [{
                src: this.avatarUrl,
                alt: this.fullName,
                authorName: this.fullName,
                dateTime: "",
                authorAvatarUrl: this.avatarUrl
            }],
            index: 0
        }));
    }

    // Удаление аватара
    public onDeleteAvatar() {
        this._request.post({
            route: ApiRoutes.deleteImage,
            data: { fileUrl: this.avatarUrl },
            successCb: () => this._dispatch(changeUserField({ field: AVATAR_URL, value: "" }))
        });
    }

    // Установка/обновление аватара
    public onSetAvatar({ id, newAvatarUrl, newPhotoUrl }: { id: string; newAvatarUrl: string; newPhotoUrl: string; }) {
        this._changeField(AVATAR_URL, newAvatarUrl);
        this._dispatch(addPhotos([{ 
            id, 
            userId: this.id, 
            path: newPhotoUrl, 
            createDate: currentDate, 
            User: { 
                id: this.id,
                firstName: this._user.firstName,
                thirdName: this._user.thirdName,
                avatarUrl: newAvatarUrl
            } 
        }]));
    }

    //-------------------------------------------------
    // Методы блока с фотографиями
    //-------------------------------------------------

    // Получить все фотографии
    public getAllPhotos(setLoading: React.Dispatch<React.SetStateAction<boolean>>) {
        this._request.get({
            route: ApiRoutes.getPhotos,
            setLoading,
            successCb: (data: { success: boolean; photos: IPhoto[]; }) => {
                this._dispatch(setPhotos(data.photos))
            }
        });
    }

    // Обновить количество фотографий
    public updatePhotosCount(count: number) {
        this._dispatch(setPhotosCount(count));
    }

    // Клик по фотографии
    public onClickPhoto(photos: IPhoto[], index: number) {
        if (photos && photos.length) {
            this._dispatch(setImagesInCarousel({
                images: photos.map(photo => ({ 
                    src: photo.path, 
                    alt: photo.id, 
                    authorName: getFullName(photo.User),
                    dateTime: photo.createDate,
                    authorAvatarUrl: photo.User?.avatarUrl || NO_PHOTO
                })),
                index
            }));
        }
    }

    // Добавление одной/нескольких фтографий
    public addPhotos(data: Object, setLoading: React.Dispatch<React.SetStateAction<boolean>>) {
        this._request.post({
            route: ApiRoutes.savePhotos,
            data,
            setLoading,
            successCb: (data: { success: boolean; photos: IPhoto[]; }) => {
                this._dispatch(addPhotos(data.photos));
            },
            config: { headers: { "Content-Type": "multipart/form-data" } }
        });
    }

    // Удаление фотографии
    public deletePhoto(data: Object, photos: IPhoto[], path: string) {
        this._request.post({
            route: ApiRoutes.deleteImage,
            data,
            successCb: () => {
                if (photos.length) {
                    const deletedPhoto = photos.find(photo => photo.path === path && photo.userId === this.id);

                    if (deletedPhoto) {
                        const indexOf = photos.indexOf(deletedPhoto);
        
                        this._dispatch(deletePhoto(indexOf));
                    }
                }
            }
        });
    }

    //-------------------------------------------------
    // Методы блока с основной информацией
    //-------------------------------------------------
    public getUserDetail(setLoading: React.Dispatch<React.SetStateAction<boolean>>) {
        this._request.get({
            route: ApiRoutes.getUserDetail, 
            setLoading,
            successCb: (data: { success: boolean, userDetail: IUserDetails }) => {
                this._dispatch(setUserDetail(data.userDetail))
            }
        });
    }

    //-------------------------------------------------
    // Методы блока с друзьями
    //-------------------------------------------------
    public getFriends(setLoading: React.Dispatch<React.SetStateAction<boolean>>) {
        this._request.get({
            route: ApiRoutes.getCountFriends, 
            setLoading,
            successCb: (data: { success: boolean, friendsCount: number, topFriends: IUser[] | null, subscribersCount: number; }) => {
                this._dispatch(setFriendsCount(data.friendsCount));
                this._dispatch(setSubscribersCount(data.subscribersCount));
                this._dispatch(setTopFriends(data.topFriends));
            }
        });
    }
}