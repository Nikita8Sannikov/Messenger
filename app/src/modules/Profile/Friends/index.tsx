import { useState, useEffect, memo } from "react";
import { useTranslation } from "react-i18next";

import List from "@mui/material/List";
import ListItem from "@mui/material/ListItem";
import ListItemAvatar from "@mui/material/ListItemAvatar";

import GridComponent from "@components/ui/Grid";
import PaperComponent from "@components/ui/Paper";
import AvatarComponent from "@components/ui/avatar";
import NoDataComponent from "@components/ui/no-data";
import SpinnerComponent from "@components/ui/spinner";
import { FriendsTab, MainFriendTabs, Pages } from "@custom-types/enums";
import { IUser } from "@custom-types/models.types";
import { useAppSelector } from "@hooks/useGlobalState";
import useProfile from "@hooks/useProfile";
import { selectMainState } from "@store/main/slice";
import { selectFriendState } from "@store/friend/slice";
import { getFullName } from "@utils/index";

import "./friends.scss";

export type onClickBlockType = (
    pathname: Pages,
    query: {
        mainTab: MainFriendTabs;
        tab?: FriendsTab;
    }
) => void;

interface IFriends {
    onlineFriends?: boolean;
    onClickBlock: onClickBlockType;
};

interface IFriendsState {
    title: string;
    count: number | string;
    users: IUser[] | null;
};

export default memo(function Friends({ onlineFriends = false, onClickBlock }: IFriends) {
    const { t } = useTranslation();

    const [loading, setLoading] = useState(false);
    const [friendsState, setFriendsState] = useState<IFriendsState>({
        title: t("profile-module.friends"),
        count: "-",
        users: null
    });

    const { topFriends, friendsCount } = useAppSelector(selectFriendState);
    const { onlineUsers } = useAppSelector(selectMainState);

    const profile = useProfile();

    // Формируем объект данных компонента
    useEffect(() => {
        const state = {
            title: onlineFriends ? t("profile-module.friends_online") : t("profile-module.friends"),
            count: onlineFriends ? onlineUsers.length : friendsCount,
            users: onlineFriends ? onlineUsers.slice(0, 6) : topFriends
        };

        setFriendsState(state);
    }, [onlineFriends, onlineUsers, topFriends, friendsCount]);

    // Получаем количество друзей, подписчиков и подгружаем первые 6 из них
    useEffect(() => {
        if (!onlineFriends) {
            profile.getFriends(setLoading);
        }
    }, []);

    // Обрабока клика по блоку
    const blockHandler = () => {
        const params: { mainTab: MainFriendTabs; tab?: FriendsTab; } = { mainTab: MainFriendTabs.allFriends };

        if (onlineFriends) {
            params.tab = FriendsTab.online;
        }

        onClickBlock(Pages.friends, params);
    };

    const { title, count, users } = friendsState;

    return <GridComponent className="friends-container__grid">
        <PaperComponent className="friends-container paper-block">
            <div className="block-title" onClick={blockHandler}>
                {title} <span className="counter">{count}</span>
            </div>

            {loading
                ? <SpinnerComponent />
                : users && users.length
                    ? <List className="friends-container__top-friends__list">
                        {users.map(user => {
                            const userName = getFullName(user);

                            return <ListItem className="friends-container__top-friends__item" key={user.id}>
                                <ListItemAvatar className="friends-container__top-friends__item__avatar-block">
                                    <AvatarComponent
                                        src={user.avatarUrl}
                                        alt={userName}
                                        isOnline={Boolean(onlineUsers.find(onlineUser => onlineUser.id === user.id))}
                                        size={50}
                                    />
                                </ListItemAvatar>

                                {user.firstName}
                            </ListItem>
                        })}
                    </List>
                    : <NoDataComponent text={t("profile-module.no_data", { isOnline: onlineFriends ? t("profile-module.online") : "" })} />
            }
        </PaperComponent>
    </GridComponent>
});