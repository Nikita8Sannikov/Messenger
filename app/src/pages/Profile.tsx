import { useNavigate } from "react-router-dom";
import Grid from "@mui/material/Grid";

import { FriendsTab, MainFriendTabs, Pages } from "@custom-types/enums";
import MainPhoto from "@modules/Profile/MainPhoto";
import Friends from "@modules/Profile/Friends";
import PersonalInfo from "@modules/Profile/PersonalInfo";
import Photos from "@modules/Profile/Photos";

import "@styles/pages/profile.scss";

export default function Profile() {
    const navigate = useNavigate();

    // Клик по названию блока
    const onClickBlock = (pathname: Pages, query: { mainTab: MainFriendTabs, tab?: FriendsTab }) => {
        // navigate({ pathname, query });
        navigate(pathname);
    };

    return <Grid container spacing={2}>
        <Grid item container xs={4} spacing={2} direction="column">
            {/* Блок моей фотографии */}
            <MainPhoto />

            {/* Блок друзей */}
            <Friends onClickBlock={onClickBlock} />

            {/* Блок друзей онлайн */}
            <Friends onClickBlock={onClickBlock} onlineFriends />
        </Grid>

        <Grid item container xs={8} spacing={2} direction="column">
            {/* Блок личной информации */}
            <PersonalInfo onClickBlock={onClickBlock} />

            {/* Блок фотографий */}
            <Photos />
        </Grid>
    </Grid>
};