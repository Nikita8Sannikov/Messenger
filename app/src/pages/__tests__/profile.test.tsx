import { act, render, screen } from "@testing-library/react";

import Profile from "@pages/Profile";
import mockProfileStore from "../../__mocks__/@store/profile";
import mockUserStore from "../../__mocks__/@store/user";

import "../../__mocks__/react-i18next";

jest.mock("@hooks/usePrepareAnotherUser", () => ({
	__esModule: true,
	default: () => ({
		isLoading: false,
		isExistProfile: true,
	}),
}));

describe("Profile Component", () => {
	beforeEach(() => {
		jest.clearAllMocks();

		const userStore = mockUserStore.getState();
		userStore.user = {
			id: "123",
			firstName: "John",
			secondName: "",
			thirdName: "Doe",
			email: "john@example.com",
			phone: "+1234567890",
			avatarUrl: "",
			avatarCreateDate: "",
			fullName: "John Doe",
		};
		userStore.userDetails = {
			birthday: "1990-01-01",
			city: "New York",
			work: "Developer",
			sex: "male",
			lastSeen: "2024-01-01",
			userId: "123",
		};

		const profileStore = mockProfileStore.getState();
		profileStore.isPrepareAnotherUser = false;
		profileStore.isMe = true;
	});

	it("matches snapshot in initial state", async () => {
		const { container } = await act(async () => {
			return render(<Profile />);
		});
		expect(container).toMatchSnapshot();
	});

	it("shows loading spinner when profile is loading", async () => {
		jest.requireMock("@hooks/usePrepareAnotherUser").default = () => ({
			isLoading: true,
			isExistProfile: false,
		});

		const { container } = await act(async () => {
			return render(<Profile />);
		});
		expect(container).toMatchSnapshot();
		expect(screen.getByTestId("spinner")).toBeInTheDocument();
	});

	it("shows loading spinner when profile does not exist", async () => {
		jest.requireMock("@hooks/usePrepareAnotherUser").default = () => ({
			isLoading: false,
			isExistProfile: false,
		});

		const { container } = await act(async () => {
			return render(<Profile />);
		});
		expect(container).toMatchSnapshot();
		expect(screen.getByTestId("spinner")).toBeInTheDocument();
	});
});
