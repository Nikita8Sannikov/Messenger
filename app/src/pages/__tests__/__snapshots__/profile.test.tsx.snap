// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component matches snapshot in initial state 1`] = `
<div>
  <div
    class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 grid-item-style css-ubb9a5-MuiGrid2-root"
  >
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 grid-item-style css-17yh78r-MuiGrid2-root"
    >
      <div
        class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 grid-item-style css-ubb9a5-MuiGrid2-root"
      >
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row main-photo css-1fzlhpv-MuiGrid2-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 main-photo__container paper-block main-photo__container__another-user css-1gtchvp-MuiPaper-root"
            style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
          >
            <div
              class="photo"
            >
              <img
                alt="user-avatar"
                class="photo__image"
                data-testid="image"
                loading="eager"
                src="/assets/images/noPhoto.jpg"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row grid-item-style css-1fzlhpv-MuiGrid2-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 friends-container paper-block css-1gtchvp-MuiPaper-root"
            style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
          >
            <div
              class="block-title"
            >
              Friends
               
              <span
                class="counter"
              >
                0
              </span>
            </div>
            <div
              class="opacity-text no-items"
              data-testid="no-data"
            >
              The friends list is empty at the moment.
            </div>
          </div>
        </div>
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row grid-item-style css-1fzlhpv-MuiGrid2-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 friends-container paper-block css-1gtchvp-MuiPaper-root"
            style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
          >
            <div
              class="block-title"
            >
              Online friends
               
              <span
                class="counter"
              >
                0
              </span>
            </div>
            <div
              class="opacity-text no-items"
              data-testid="no-data"
            >
              На данный момент список онлайн друзей пуст.
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-8 grid-item-style css-1cdctrw-MuiGrid2-root"
    >
      <div
        class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 grid-item-style css-ubb9a5-MuiGrid2-root"
      >
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row grid-item-style css-1fzlhpv-MuiGrid2-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 info-container paper-block css-1gtchvp-MuiPaper-root"
            style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
          >
            <div
              class="info-container__main-info"
            >
              <div
                class="info-container__username"
              >
                John Doe
              </div>
            </div>
            <div
              class="info-container__short-info-block"
            >
              <div
                class="info-container__short-info-block__row"
              >
                <div
                  class="info-container__short-info-block__row__title"
                >
                  Birthday
                  :
                </div>
                <span
                  class="info-container__short-info-block__row__value"
                >
                  1990-01-01
                </span>
              </div>
              <div
                class="info-container__short-info-block__row"
              >
                <div
                  class="info-container__short-info-block__row__title"
                >
                  City
                  :
                </div>
                <span
                  class="info-container__short-info-block__row__value"
                >
                  New York
                </span>
              </div>
              <div
                class="info-container__short-info-block__row"
              >
                <div
                  class="info-container__short-info-block__row__title"
                >
                  Place of work
                  :
                </div>
                <span
                  class="info-container__short-info-block__row__value"
                >
                  Developer
                </span>
              </div>
            </div>
            <div
              class="info-container__counts-block"
            >
              <div
                class="info-container__counts-block__count"
              >
                <span
                  class="info-container__counts-block__count__value"
                >
                  0
                </span>
                 
              </div>
              <div
                class="info-container__counts-block__count"
              >
                <span
                  class="info-container__counts-block__count__value"
                >
                  0
                </span>
                 
              </div>
              <div
                class="info-container__counts-block__count"
              >
                <span
                  class="info-container__counts-block__count__value"
                >
                  0
                </span>
                 
              </div>
              <div
                class="info-container__counts-block__count"
              >
                <span
                  class="info-container__counts-block__count__value"
                >
                  0
                </span>
                 
              </div>
              <div
                class="info-container__counts-block__count"
              >
                <span
                  class="info-container__counts-block__count__value"
                >
                  0
                </span>
                 
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid2-root MuiGrid2-direction-xs-row grid-item-style css-1fzlhpv-MuiGrid2-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 photo-container paper-block css-1gtchvp-MuiPaper-root"
            style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
          >
            <div
              class="photo-container__title"
            >
              <div
                class="block-title photo-container__title__text"
              >
                Photos
                <span
                  class="counter"
                >
                  0
                </span>
              </div>
              <a
                class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineHover photo-container__title__link css-jygu7u-MuiTypography-root-MuiLink-root"
                data-testid="link"
              >
                Watch all
              </a>
            </div>
            <div
              class="opacity-text no-items"
              data-testid="no-data"
            >
              No photos
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Profile Component shows loading spinner when profile does not exist 1`] = `
<div>
  <div
    class="user-profile-page-spinner"
  >
    <div
      class="spinner"
      data-testid="spinner"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1wxecgq-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-19t5dcl-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;

exports[`Profile Component shows loading spinner when profile is loading 1`] = `
<div>
  <div
    class="user-profile-page-spinner"
  >
    <div
      class="spinner"
      data-testid="spinner"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1wxecgq-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-19t5dcl-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
